app/src/main/java/com/edulectronics/tinycircuit/ui/CircuitActivity.java

```diff
@@ -66,20 +77,66 @@ public boolean onTouch(View v, MotionEvent event) {
 
     /*This code adds a menu to the side*/
     private void makeMenu(){
-        ListView menu = (ListView) findViewById(R.id.menu);
-        String[] items = getResources().getStringArray(R.array.items);
-        ArrayAdapter<String> adapter = new ArrayAdapter<>(
-                this, android.R.layout.simple_list_item_1, items);
-        menu.setAdapter(adapter);
-
-        menu.setOnItemClickListener(new AdapterView.OnItemClickListener() {
-            public void onItemClick(AdapterView<?> adapterView, View view,
-                                    int position, long itemId){
-                /*Not implemented, should select desired item*/
+        ExpandableListView expandableList = (ExpandableListView) findViewById(R.id.expandablelist);
+        makeLists();
+
+        ExpandableListAdapter adapter = new ExpandableListAdapter(
+                this, headers, children, expandableList);
+        expandableList.setAdapter(adapter);
+
+        expandableList.setOnChildClickListener(new ExpandableListView.OnChildClickListener() {
+            @Override
+            public boolean onChildClick(ExpandableListView expandableListView, View view,
+                                        int groupPosition, int childPosition, long rowID) {
+                /*Handle click event of child over here*/
+                return false;
+            }
+        });
+        expandableList.setOnGroupClickListener(new ExpandableListView.OnGroupClickListener() {
+            @Override
+            public boolean onGroupClick(ExpandableListView expandableListView, View view,
+                                        int groupPosition, long rowID) {
+                /*Handle click event of group over here*/
+                return false;
             }
         });
     }
 
+    /*Makes the groups and children*/
+    private void makeLists() {
```

Comment:
Ik denk dat ik een aparte class menu ga maken die het menu maakt met een gegeven lijst van componenten. Voor nu is dit goed